### PR TITLE
Handle image_embeds in ViltModel

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Handle `image_embeds` in `ViltModel` / `ViltForImagesAndTextClassification`.

(looks like `Vilt` is the first model introducing `image_embeds` argument.)

## More Info

The `image_embeds` in `ViltForImagesAndTextClassification` should have `num_images` dimension as `pixel_values` has, as far as I understand.

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
LGTM, but let's wait for @NielsRogge final review before merging!

File: src/transformers/models/vilt/modeling_vilt.py [hunk 1]
```diff
@@ -811,12 +811,16 @@ def forward(
...
-            raise ValueError("You have to specify pixel_values")
+        if pixel_values is not None and image_embeds is not None:
+            raise ValueError("You cannot specify both pixel_values and image_embeds at the same time")
+        elif pixel_values is None and image_embeds is None:
+            raise ValueError("You have to specify either pixel_values or image_embeds")
 
-        batch_size, num_channels, height, width = pixel_values.shape
+        image_batch_size = pixel_values.shape[0] if pixel_values is not None else image_embeds.shape[0]
+        if image_batch_size != batch_size:
+            raise ValueError("The text inputs and image inputs need to have the same batch size")
```
Comment (NielsRogge) [inline src/transformers/models/vilt/modeling_vilt.py:821]: Maybe for consistency, we can call the other one `text_batch_size` (instead of `batch_size`).

**Review by NielsRogge — APPROVED** [review]
Thanks for improving this!

Out of interest: were you experimenting with ViLT?

**Comment by ydshieh** [comment]
> Thanks for improving this!
> 
> Out of interest: were you experimenting with ViLT?

Not for this PR. I tried to fix a CI (vit-mae), which was about `test_torchscript`.
It turns out to be related model main input -> I worked/improved on it -> more models involved including ViLT -> I just took this chance to work on this PR (otherwise I would forget it very quickly)

File: src/transformers/models/vilt/modeling_vilt.py (hunk 1 above)
Comment (ydshieh) [reply src/transformers/models/vilt/modeling_vilt.py]: done :-) & wait for green CI

**Pull request merged**